### PR TITLE
hp-ux port

### PR DESCRIPTION
**Pull request by malexzx**
following patch allows  the mongo-c driver to work on HP-UX 11.31 ia64 platform

To compile on HPUX we need:

gcc-4.8.5
pass -D_REENTRANT -D_INCLUDE_HPUX_SOURCE to CFLAGS
pass -mlp64 to CFLAGS and LDFLAGS for 64bit version

### CODE REVIEW

File: src/bson/bson-macros.h [hunk 1]
```diff
@@ -77,6 +77,10 @@
 #  define BSON_API
 #endif
 
+#ifdef __hpux__
+#undef MIN
+#undef MAX
```
Comment (bjori) [inline src/bson/bson-macros.h]: Why did you need to do this?
These macros seem to work as expected

Comment (malexzx) [inline src/bson/bson-macros.h]: Not need now.
Renaming these macros in mongo-c-driver will solve this issue.

**Comment by bjori** [comment]
Merged as https://github.com/mongodb/libbson/commit/96bccf61a4bd60b4456d14e2924c4ba93df2dd78

Thanks for the patch!
